### PR TITLE
chore(service): add rqmetrics to deployment

### PR DESCRIPTION
**Pull request by jsam**
closes: #1419

TODO:
- merge on https://github.com/SwissDataScienceCenter/rqmetrics/pull/4

### CODE REVIEW

File: helm-chart/renku-core/templates/deployment.yaml [hunk 1]
```diff
@@ -27,6 +27,13 @@ spec:
         - name: shared-volume
           emptyDir: {}
       containers:
+        - name: {{ .Chart.Name}}-rqmetrics
+          image: "justsam/rqmetrics:latest"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: RQ_REDIS_HOST
+              value: {{ include "call-nested" (list . "redis" "redis.fullname") }}-master
+
```
Comment (rokroskar) [inline helm-chart/renku-core/templates/deployment.yaml:36]: does this need annotations to be picked up by prometheus?

Comment (rokroskar) [reply helm-chart/renku-core/templates/deployment.yaml:36]: actually I guess they're needed in the service: https://github.com/SwissDataScienceCenter/renku-graph/blob/master/helm-chart/renku-graph/templates/event-log-service.yaml#L10-L13

File: helm-chart/renku-core/templates/deployment.yaml [hunk 2]
```diff
@@ -27,6 +31,13 @@ spec:
         - name: shared-volume
           emptyDir: {}
       containers:
+        - name: {{ .Chart.Name}}-rqmetrics
+          image: "justsam/rqmetrics:latest"
```
Comment (rokroskar) [inline helm-chart/renku-core/templates/deployment.yaml:35]: This needs to be templated, e.g.

```
image: {{ .Values.metrics.image }}
```

File: helm-chart/renku-core/templates/deployment.yaml [hunk 3]
```diff
@@ -27,6 +31,13 @@ spec:
...
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
```
Comment (rokroskar) [inline helm-chart/renku-core/templates/deployment.yaml:36]:
```suggestion
          imagePullPolicy: {{ .Values.metrics.image.pullPolicy }}
```

File: helm-chart/renku-core/templates/deployment.yaml [hunk 4]
```diff
@@ -27,6 +31,13 @@ spec:
         - name: shared-volume
           emptyDir: {}
       containers:
+        - name: {{ .Chart.Name}}-rqmetrics
```
Comment (rokroskar) [inline helm-chart/renku-core/templates/deployment.yaml:34]: This should be in an `if` block, e.g.

```
{{ if .Values.metrics.enabled }}
```

**Review by rokroskar — CHANGES_REQUESTED** [review]

File: helm-chart/renku-core/templates/deployment.yaml [hunk 5]
```diff
@@ -14,6 +14,10 @@ spec:
       app.kubernetes.io/instance: {{ .Release.Name }}
   template:
     metadata:
+      annotations:
+        prometheus.io/scrape: 'true'
+        prometheus.io/path: '/metrics'
+        prometheus.io/port: '9005'
```
Comment (rokroskar) [inline helm-chart/renku-core/templates/deployment.yaml:20]:
```suggestion
        prometheus.io/port: '8765'
```

**Review by rokroskar — CHANGES_REQUESTED** [review]

**Review by Panaetius — APPROVED** [review]
🚀

File: helm-chart/renku-core/values.yaml [hunk 6]
```diff
@@ -31,6 +31,13 @@ service:
 ingress:
   enabled: false
 
+metrics:
+  enabled: true
```
Comment (rokroskar) [inline helm-chart/renku-core/values.yaml:35]: usually the convention is to have metrics exporting default to false - can we do the same here?

**Review by rokroskar — CHANGES_REQUESTED** [review]

**Review by rokroskar — APPROVED** [review]
Thanks @jsam, nothing quite as satisfying as watching a dashboard do its thing! :)

**Pull request merged**